### PR TITLE
Logging++

### PR DESCRIPTION
**Pull request by facontidavide**
Hi,

this  PR add many capabilities in terms of loggin, in particular.

1) **Log to File** + command line tool to print it (**bt_log_cat**)
2) Log as **minitrace** (see https://github.com/hrydgard/minitrace )
3) Publish on TCP woth ZeroMQ + a command line tool similar to rosbag record (**bt_record**).

To get a feeling of what this logging can do, take a look at the latest videos

https://vimeo.com/album/5201743

Cheers

### CODE REVIEW

**Comment by facontidavide** [comment]
If you have any question, just ask... ;)

File: src/bt_zmq_publisher.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
+#include "behavior_tree_logger/bt_zmq_publisher.h"
+#include "behavior_tree_logger/bt_flatbuffer_helper.h"
+#include <future>
+
+namespace BT {
+
+void PublisherZMQ::createStatusBuffer()
+{
+    _status_buffer.clear();
```
Comment (miccol) [inline src/bt_zmq_publisher.cpp]: why the underscore is a prefix rathern than a suffix?

Comment (facontidavide) [reply src/bt_zmq_publisher.cpp]: Whoops

**Comment by facontidavide** [comment]
Fixed the underscore thing

**Pull request merged**